forem/forem
# Remove fix-db-schema-conflicts

## What type of PR is this? (check all applicable)


## Description

So far we used the [`fix-db-schema-conflicts`](https://github.com/jakeonrails/fix-db-schema-conflicts) gem in development, which alphabetically sorts columns in the generated `schema.rb` file to avoid conflicts. Alas, this gem has been unmaintained for a while, which is slowly becoming a bit of a burden:

1. We had to fork it to make it work with Ruby 3.
2. Its Rubocop config contains an outdated cop that generates a warning after every migration:
    ```ruby
    Error: The `Layout/AlignHash` cop has been renamed to `Layout/HashAlignment`.
(obsolete configuration found in vendor/cache/ruby/2.7.0/cache/ruby/2.7.0/gems/fix-db-schema-conflicts-3.0.3/.rubocop_schema.53.yml, please update it)
    ```

While browsing the documentation for [`strong_migrations`]() while reviewing another PR, I noticed that it can also be used to alphabetically sort the columns, so I decided to get rid of an inconvenient dependency.

https://github.com/ankane/strong_migrations#schema-sanity

## Related Tickets & Documents

n/a

## QA Instructions, Screenshots, Recordings

1. Manually mess up the column order
2. Add a new migration
3. Run it and ensure that all columns are alphabetically sorted again

### UI accessibility concerns?

n/a

## Added tests?

- [X] No, and this is why: doesn't affect app code

## Added to documentation?
